### PR TITLE
Dala 3173 speed up framework search

### PR DESCRIPTION
**Pull request by fsoffel**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [ ] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] The version of main currently active on prod is deployed to a dev server with `Reset non-user related Docker Volumes & Re-populate` turned on
  - [ ] It's verified that the CD run is green
  - [ ] The data from prod is migrated using `./migrateData.sh dataland.com <SOURCE_API_KEY> <TARGET> <TARGET_API_KEY>` 
  - [ ] The feature branch is deployed to the same server with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the dev server
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server
- [ ] Make sure that pull requests in the development tools repository, which are connected to this pull requests, 
are merged

### CODE REVIEW

File: dataland-backend/src/main/resources/application-development.properties [hunk 1]
```diff
@@ -1,7 +1,7 @@
 spring.security.oauth2.resourceserver.jwt.issuer-uri=https://local-dev.dataland.com/keycloak/realms/datalandsecurity
 spring.security.oauth2.resourceserver.jwt.jwk-set-uri=https://local-dev.dataland.com/keycloak/realms/datalandsecurity/protocol/openid-connect/certs
 spring.datasource.url=jdbc:postgresql://127.0.0.1:5433/backend
-spring.jpa.hibernate.ddl-auto=create
+spring.jpa.hibernate.ddl-auto=update
```
Comment (stewal19) [inline dataland-backend/src/main/resources/application-development.properties:4]: Why this change?

File: dataland-frontend/src/utils/SearchCompaniesForFrameworkDataPageDataRequester.ts [hunk 2]
```diff
@@ -5,69 +5,22 @@
...
-
 export interface FrameworkDataSearchFilterInterface {
   companyNameFilter: string;
   frameworkFilter: Array<DataTypeEnum>;
   countryCodeFilter: Array<string>;
   sectorFilter: Array<string>;
 }
 
-/**
- * Retrieve the value of the Perm Id of a company. Throws an exception if no perm id is found
```
Comment (stewal19) [inline dataland-frontend/src/utils/SearchCompaniesForFrameworkDataPageDataRequester.ts:34]: Gone, because the return of the modified endpoint gives those information directly?

File: dataland-frontend/tests/component/components/resources/frameworkDataSearch/FrameworkDataSearchResults.cy.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
 import FrameworkDataSearchResults from "@/components/resources/frameworkDataSearch/FrameworkDataSearchResults.vue";
 import { minimalKeycloakMock } from "@ct/testUtils/Keycloak";
-import { type DataSearchStoredCompany } from "@/utils/SearchCompaniesForFrameworkDataPageDataRequester";
+import {BasicCompanyInformation} from "../../../../../build/clients/backend";
```
Comment (stewal19) [inline dataland-frontend/tests/component/components/resources/frameworkDataSearch/FrameworkDataSearchResults.cy.ts:3]: import statement should be something like "@"

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/utils/ApiAccessor.kt [hunk 4]
```diff
@@ -9,22 +9,7 @@ import org.dataland.datalandbackend.openApiClient.api.MetaDataControllerApi
...
-import org.dataland.datalandbackend.openApiClient.model.DataMetaInformation
-import org.dataland.datalandbackend.openApiClient.model.DataTypeEnum
-import org.dataland.datalandbackend.openApiClient.model.EuTaxonomyDataForFinancials
-import org.dataland.datalandbackend.openApiClient.model.EuTaxonomyDataForNonFinancials
-import org.dataland.datalandbackend.openApiClient.model.LksgData
-import org.dataland.datalandbackend.openApiClient.model.PathwaysToParisData
-import org.dataland.datalandbackend.openApiClient.model.SfdrData
-import org.dataland.datalandbackend.openApiClient.model.SmeData
-import org.dataland.datalandbackend.openApiClient.model.StoredCompany
+import org.dataland.datalandbackend.openApiClient.model.*
```
Comment (stewal19) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/utils/ApiAccessor.kt:12]: linter will propably not like this

File: dataland-frontend/tests/component/components/resources/frameworkDataSearch/FrameworkDataSearchResults.cy.ts (hunk 3 above)
Comment (BenediktLaubmann) [reply dataland-frontend/tests/component/components/resources/frameworkDataSearch/FrameworkDataSearchResults.cy.ts]: fixed

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/UserApi.kt [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+
+/**
+ * Defines the restful dataland-backend API regarding user data exchange
+ */
+@RequestMapping("/user")
+@SecurityRequirement(name = "default-bearer-auth")
+@SecurityRequirement(name = "default-oauth")
+interface UserApi {
```
Comment (stewal19) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/UserApi.kt:20]: Name feels a bit weird.  Seeing this name i would expect to get some interaction with the user data stored on dataland. I would not expect to get company information

File: dataland-backend/src/main/resources/application-development.properties (hunk 1 above)
Comment (BenediktLaubmann) [reply dataland-backend/src/main/resources/application-development.properties:4]: will be reverted

File: dataland-frontend/src/utils/SearchCompaniesForFrameworkDataPageDataRequester.ts (hunk 2 above)
Comment (BenediktLaubmann) [reply dataland-frontend/src/utils/SearchCompaniesForFrameworkDataPageDataRequester.ts:34]: yes

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/UserApi.kt (hunk 5 above)
Comment (BenediktLaubmann) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/UserApi.kt:20]: will change to "UserUploadsApi"

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/UserController.kt [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import org.dataland.keycloakAdapter.auth.DatalandAuthentication
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.RestController
+
+/**
+ * Defines the restful dataland-backend API regarding user data exchange
+ */
+@RestController
+class UserController(
```
[the remark this answers is not in the document]
Comment (BenediktLaubmann) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/UserController.kt:16]: will change to "UserUploadsController"

[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/UserController.kt]: ok

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/UserApi.kt (hunk 5 above)
Comment (stewal19) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/UserApi.kt]: ok

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/UserUploadsApi.kt [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.http.ResponseEntity
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+
+/**
+ * Defines the restful dataland-backend API regarding user data exchange
+ */
+@RequestMapping("/users")
```
Comment (stewal19) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/UserUploadsApi.kt:17]: Isn't this endpoint similar to the meta-data endpoint? Why this new path /users?
I for one would expect to get somehow user information like how many users, userIds and so on, under this route.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/BasicCompanyInformation.kt [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+    val companyName: String
+
+    val headquarters: String
+
+    val countryCode: String
+
+    val sector: String?
+
+    var permId: String?
```
Comment (stewal19) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/BasicCompanyInformation.kt:17]: What about other identifiers like isin, lei, etc?

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/DatasetMetaInfoEntityForMyDatasets.kt [hunk 9]
```diff
@@ -0,0 +1,16 @@
+package org.dataland.datalandbackend.entities
+
+
+/**
+ * A variation fo DataMetaInformationForMyDatasets for database interaction
```
Comment (stewal19) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/DatasetMetaInfoEntityForMyDatasets.kt:5]: Typo

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerGetCompaniesEndpointTest.kt [hunk 10]
```diff
@@ -1,10 +1,6 @@
 package org.dataland.e2etests.tests
 
-import org.dataland.datalandbackend.openApiClient.model.CompanyInformation
-import org.dataland.datalandbackend.openApiClient.model.DataMetaInformation
-import org.dataland.datalandbackend.openApiClient.model.IdentifierType
-import org.dataland.datalandbackend.openApiClient.model.QaStatus
-import org.dataland.datalandbackend.openApiClient.model.StoredCompany
+import org.dataland.datalandbackend.openApiClient.model.*
```
Comment (stewal19) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerGetCompaniesEndpointTest.kt:3]: Is the linter fine with this?

File: dataland-frontend/src/utils/SearchCompaniesForFrameworkDataPageDataRequester.ts (hunk 2 above)
Comment (stewal19) [reply dataland-frontend/src/utils/SearchCompaniesForFrameworkDataPageDataRequester.ts:34]: thank you

File: dataland-frontend/tests/component/utils/SearchCompaniesForFrameworkDataPageDataRequester.cy.ts [hunk 11]
```diff
@@ -1,80 +0,0 @@
-import { QaStatus, type DataTypeEnum, type StoredCompany, IdentifierType } from "@clients/backend";
-import type Keycloak from "keycloak-js";
-import { getCompanyDataForFrameworkDataSearchPage } from "@/utils/SearchCompaniesForFrameworkDataPageDataRequester";
```
Comment (stewal19) [inline dataland-frontend/tests/component/utils/SearchCompaniesForFrameworkDataPageDataRequester.cy.ts:3]: Why is this test no longer necessary?

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/DatasetMetaInfoEntityForMyDatasets.kt (hunk 9 above)
Comment (BenediktLaubmann) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/DatasetMetaInfoEntityForMyDatasets.kt:5]: fixed

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/ressources/StoredCompaniesRepositoriesTest.kt [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+            uploadTime = 2,
+            reportingPeriod = "2",
+            currentlyActive = null,
+            qaStatus = QaStatus.Pending
+        )
+        )
+
+
+
+        println(companyQueryManager.getCompanyById(acceptedCompany.companyId))
```
Comment (stewal19) [inline dataland-backend/src/test/kotlin/org/dataland/datalandbackend/ressources/StoredCompaniesRepositoriesTest.kt:92]: Debugging statement?

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/ressources/StoredCompaniesRepositoriesTest.kt [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+        val testQueryResult = companyQueryManager.searchCompaniesAndGetApiModel(
+            StoredCompanySearchFilter(
+                countryCodeFilter = emptyList(),
+                dataTypeFilter = emptyList(),
+                sectorFilter = emptyList(),
+                searchString = ""
+            )
+        )
+        println(testQueryResult)
```
Comment (stewal19) [inline dataland-backend/src/test/kotlin/org/dataland/datalandbackend/ressources/StoredCompaniesRepositoriesTest.kt:101]: Debugging statement?

File: dataland-frontend/tests/component/components/resources/frameworkDataSearch/FrameworkDataSearchResults.cy.ts (hunk 3 above)
Comment (stewal19) [reply dataland-frontend/tests/component/components/resources/frameworkDataSearch/FrameworkDataSearchResults.cy.ts]: ok

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/BasicCompanyInformation.kt (hunk 8 above)
Comment (fsoffel) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/BasicCompanyInformation.kt:17]: They are not shown on the /companies page. This only contains data for this page.

File: dataland-frontend/tests/component/utils/SearchCompaniesForFrameworkDataPageDataRequester.cy.ts (hunk 11 above)
Comment (fsoffel) [reply dataland-frontend/tests/component/utils/SearchCompaniesForFrameworkDataPageDataRequester.cy.ts:3]: Because the removing of companies with non accepted datasets only is now done already in the queue

Comment (fsoffel) [reply dataland-frontend/tests/component/utils/SearchCompaniesForFrameworkDataPageDataRequester.cy.ts:3]: I added a new backend e2e test to test for this.

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/ressources/StoredCompaniesRepositoriesTest.kt (hunk 12 above)
Comment (fsoffel) [reply dataland-backend/src/test/kotlin/org/dataland/datalandbackend/ressources/StoredCompaniesRepositoriesTest.kt]: Yes. The file is gone now and the intent is implemented in an e2e test.

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/ressources/StoredCompaniesRepositoriesTest.kt (hunk 13 above)
Comment (fsoffel) [reply dataland-backend/src/test/kotlin/org/dataland/datalandbackend/ressources/StoredCompaniesRepositoriesTest.kt]: Same as above

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/BasicCompanyInformation.kt (hunk 8 above)
Comment (stewal19) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/BasicCompanyInformation.kt:17]: ok, if in the future other identifiers should be displayed, this would have to be extended.

File: dataland-frontend/tests/component/utils/SearchCompaniesForFrameworkDataPageDataRequester.cy.ts (hunk 11 above)
Comment (stewal19) [reply dataland-frontend/tests/component/utils/SearchCompaniesForFrameworkDataPageDataRequester.cy.ts:3]: ok

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerGetCompaniesEndpointTest.kt (hunk 10 above)
Comment (BenediktLaubmann) [reply dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerGetCompaniesEndpointTest.kt]: It should not be but since this file lies not in in the frontend, the linter does not "see" it. 
I changed my IDE settings and now these wildcard-imports are not possible anymore.
Also changed this import.

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/ressources/StoredCompaniesRepositoriesTest.kt (hunk 12 above)
Comment (stewal19) [reply dataland-backend/src/test/kotlin/org/dataland/datalandbackend/ressources/StoredCompaniesRepositoriesTest.kt]: ok

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerGetCompaniesEndpointTest.kt (hunk 10 above)
Comment (stewal19) [reply dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerGetCompaniesEndpointTest.kt]: ok

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/UserUploadsApi.kt (hunk 7 above)
Comment (fsoffel) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/UserUploadsApi.kt:17]: Hmm, I see your point. I am not sure about the perfect way to handle this. However, these endpoints are not returning the usual meta infos but a combination with some company data specifically tailored for the "My Datasets" page. This felt like the natural way of managing this resource because it is identified with a user.

Comment (stewal19) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/UserUploadsApi.kt:17]: ok

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/repositories/DataMetaInformationRepository.kt [hunk 14]
```diff
@@ -69,4 +70,32 @@ interface DataMetaInformationRepository : JpaRepository<DataMetaInformationEntit
...
+            " datainfo.upload_time as uploadTime" +
+            " from (" +
+            " SELECT company_id, data_id, data_type, reporting_period, quality_status, currently_active, upload_time " +
+            " from data_meta_information meta" +
+            " where uploader_user_id = :userId" +
+            " ) datainfo" +
+            " LEFT JOIN stored_companies company" +
+            " ON company.company_id = datainfo.company_id",
+    )
+    fun getUserDataMetaInfos(
```
Comment (stewal19) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/repositories/DataMetaInformationRepository.kt:98]: should this be called then getUserUploadsDataMetaInfos, as we not get metadata of the user but other of uploaded data?

Comment (stewal19) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/repositories/DataMetaInformationRepository.kt]: @fsoffel @BenediktLaubmann  could you also have a look at this?

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/DataMetaInformationForMyDatasets.kt [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+    val reportingPeriod: String,
+    val qualityStatus: QaStatus,
+    val currentlyActive: Boolean?,
+    var uploadTime: Long,
+) {
+    companion object {
+        /**
+         * A function to construct a DatametaInformationForMyDatasets object from a entity object
+         */
+        fun fromEntity(entity: DatasetMetaInfoEntityForMyDatasets) = DataMetaInformationForMyDatasets(
```
Comment (stewal19) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/DataMetaInformationForMyDatasets.kt:23]: Name could be more speaking i think?

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/repositories/DataMetaInformationRepository.kt (hunk 14 above)
Comment (stewal19) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/repositories/DataMetaInformationRepository.kt]: ok was already adressed. closed

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataMetaInformationManager.kt [hunk 16]
```diff
@@ -91,17 +92,23 @@ class DataMetaInformationManager(
     }
 
     /**
-     * Method to make the data manager search for meta info
-     * @param companyId if not empty, it filters the requested meta info to a specific company
-     * @param dataType if not empty, it filters the requested meta info to a specific data type
-     * @param reportingPeriod if not empty, it filters the requested meta info to a specific reporting period
-     * @param showOnlyActive if true, it will only return datasets marked "active"
-     * @return a list of meta info about data depending on the filters
+     * Method to make delete the data meta information for a given dataId
```
Comment (stewal19) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataMetaInformationManager.kt:95]: Doc sounds strange

Comment (fsoffel) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataMetaInformationManager.kt:95]: Will be pushed in 2 minutes

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/utils/ApiAccessor.kt (hunk 4 above)
Comment (fsoffel) [reply dataland-e2etests/src/test/kotlin/org/dataland/e2etests/utils/ApiAccessor.kt]: done

File: dataland-backend/src/main/resources/application-development.properties (hunk 1 above)
Comment (fsoffel) [reply dataland-backend/src/main/resources/application-development.properties]: done

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/DataMetaInformationForMyDatasets.kt (hunk 15 above)
Comment (fsoffel) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/DataMetaInformationForMyDatasets.kt:23]: Done

File: dataland-backend/src/main/resources/application-development.properties (hunk 1 above)
Comment (stewal19) [reply dataland-backend/src/main/resources/application-development.properties]: ok

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/utils/ApiAccessor.kt (hunk 4 above)
Comment (stewal19) [reply dataland-e2etests/src/test/kotlin/org/dataland/e2etests/utils/ApiAccessor.kt]: ok

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataMetaInformationManager.kt (hunk 16 above)
Comment (stewal19) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataMetaInformationManager.kt]: thank you for correcting the whole doc for the delete function!

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/DataMetaInformationForMyDatasets.kt (hunk 15 above)
Comment (stewal19) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/DataMetaInformationForMyDatasets.kt]: ok
